sympy/solvers/ode/tests/test_single.py

```diff
@@ -293,33 +293,6 @@ def test_factorable():
     _ode_solver_test(_get_examples_ode_sol_factorable())
 
 
-def test_issue_15889():
-    eq = exp(f(x).diff(x))-f(x)**2
-    sol = Eq(NonElementaryIntegral(1/log(y**2), (y, f(x))), C1 + x)
-    assert sol.dummy_eq(dsolve(eq))
-    assert checkodesol(eq, sol) == (True, 0)
```

Comment:
@oscarbenjamin can you help how should this example be handled?

Reply:
I guess that if the expected solution (i.e. `sol` above) has any `Dummy` symbols we can use `dummy_eq` to compare the solutions:
```
In [63]: eq = exp(f(x).diff(x))-f(x)**2                                                                                           

In [64]: sol = dsolve(eq, f(x))                                                                                                   

In [65]: sol.has(Dummy)                                                                                                           
Out[65]: True
```
We should check for Dummy in the expected solution though, *not* the result from `dsolve`.

Reply:
when we look at expected solution `eq = Eq(NonElementaryIntegral(1/log(y**2), (y, f(x))), C1 + x)` so here `eq.has(Dummy)` will be false.

should i define a dummy variable and write the equation?

Reply:
> should i define a dummy variable and write the equation?

Yes. The convention can be that using a `Dummy` in the "expected" solution means it should be compared using `dummy_eq` in the tests.